packages/react-instantsearch-core/src/core/indexUtils.js

```diff
@@ -161,44 +156,84 @@ function getNamespaceAndAttributeName(id) {
   return { namespace, attributeName };
 }
 
-// eslint-disable-next-line max-params
+function hasRefinements({
+  multiIndex,
+  indexId,
+  namespace,
+  attributeName,
+  id,
+  searchState,
+}) {
+  if (multiIndex && namespace) {
+    return (
+      searchState.indices &&
+      searchState.indices[indexId] &&
+      searchState.indices[indexId][namespace] &&
+      searchState.indices[indexId][namespace].hasOwnProperty(attributeName)
+    );
+  }
+
+  if (multiIndex) {
+    return (
+      searchState.indices &&
+      searchState.indices[indexId] &&
+      searchState.indices[indexId].hasOwnProperty(id)
+    );
+  }
+
+  if (namespace) {
+    return (
+      searchState[namespace] &&
+      searchState[namespace].hasOwnProperty(attributeName)
+    );
+  }
+
+  return searchState.hasOwnProperty(id);
+}
+
+function getRefinements({
+  multiIndex,
+  indexId,
+  namespace,
+  attributeName,
+  id,
+  searchState,
+}) {
+  if (multiIndex && namespace) {
+    return searchState.indices[indexId][namespace][attributeName];
+  }
+  if (multiIndex) {
+    return searchState.indices[indexId][id];
+  }
+  if (namespace) {
+    return searchState[namespace][attributeName];
+  }
+
+  return searchState[id];
+}
+
 export function getCurrentRefinementValue(
   props,
   searchState,
   context,
   id,
-  defaultValue,
-  refinementsCallback = x => x
+  defaultValue
 ) {
   const indexId = getIndexId(context);
   const { namespace, attributeName } = getNamespaceAndAttributeName(id);
-  const refinements =
-    (hasMultipleIndices(context) &&
-      searchState.indices &&
-      namespace &&
-      searchState.indices[`${indexId}`] &&
-      has(searchState.indices[`${indexId}`][namespace], `${attributeName}`)) ||
-    (hasMultipleIndices(context) &&
-      searchState.indices &&
-      has(searchState, `indices.${indexId}.${id}`)) ||
-    (!hasMultipleIndices(context) &&
-      namespace &&
-      has(searchState[namespace], attributeName)) ||
-    (!hasMultipleIndices(context) && has(searchState, id));
-  if (refinements) {
-    let currentRefinement;
-
-    if (hasMultipleIndices(context)) {
-      currentRefinement = namespace
-        ? get(searchState.indices[`${indexId}`][namespace], attributeName)
-        : get(searchState.indices[indexId], id);
-    } else {
-      currentRefinement = namespace
-        ? get(searchState[namespace], attributeName)
-        : get(searchState, id);
-    }
+  const multiIndex = hasMultipleIndices(context);
+  const args = {
```

Comment:
IMO it was easier with the value inlined inside the call.

Reply:
was mainly  to keep them in the same order